app/src/main/java/org/xbmc/kore/ui/BaseActivity.java

```diff
@@ -15,33 +15,50 @@
  */
 package org.xbmc.kore.ui;
 
+import android.content.Intent;
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.preference.PreferenceManager;
 import android.support.v7.app.AppCompatActivity;
+import android.view.KeyEvent;
 
 import org.xbmc.kore.Settings;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.method.Application;
+import org.xbmc.kore.jsonrpc.type.GlobalType;
+import org.xbmc.kore.ui.sections.hosts.AddHostActivity;
 import org.xbmc.kore.utils.UIUtils;
 
 /**
  * Base activity, where common behaviour is implemented
  */
 public abstract class BaseActivity extends AppCompatActivity {
 
-	@Override
-	protected void onCreate(Bundle savedInstanceState) {
+    /**
+     * Host manager singleton
+     */
+    protected HostManager hostManager = null;
```

Comment:
I think that each specific activity creates (gets) its hostManager, instead of relying on a protected member inherited from the superclass. It's not obvious when reading the code where it comes from.

Reply:
I've made the changes, because you were making a valid point with the inheritance depth growth, but if I get it right the point you are trying to make here is not something that applies here in my opinion.

To me as soon as I knew that `HostManager` is a singleton, I've stopped caring about where it comes from, in fact I'd actually expect it to be held somewhere "deep", like the `BaseActivity`. I guess maybe if you are looking to go away from the singleton pattern in the future for the `HostManager` then I can get behind what you were saying.